interface/batchcom/emailnotification.php

```diff
@@ -28,52 +28,44 @@
     exit();
 }
 
- // default value
-$next_app_date = date("Y-m-d");
-$hour = "12";
-$min = "15";
-$provider_name = "EMR Group";
-$message = "Welcome to EMR Group";
-$type = "Email";
-$email_sender = "EMR Group";
-$email_subject = "Welcome to EMR Group";
 // process form
 if ($_POST['form_action'] == 'save') {
     if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
         CsrfUtils::csrfNotVerified();
     }
 
-    //validation uses the functions in notification.inc.php
-    if ($_POST['email_sender'] == "") {
-        $form_err .= xl('Empty value in "Email Sender"') . '<br />';
+    if (! is_numeric($_POST['notification_id'])) {  // shouldn't happen
+        $form_err .= xl('Missing/invalid notification id') . '<br />';
     }
 
-    if ($_POST['email_subject'] == "") {
-        $form_err .= xl('Empty value in "Email Subject"') . '<br />';
+    if (empty($_POST['email_sender'])) {
+        $form_err .= xl('Empty value in "Email Sender"') . '<br />';
     }
 
-    //validate dates
-    if (!check_date_format($_POST['next_app_date'])) {
-        $form_err .= xl('Date format for "Next Appointment" is not valid') . '<br />';
+    if (empty($_POST['email_subject'])) {
+        $form_err .= xl('Empty value in "Email Subject"') . '<br />';
     }
 
-    // validates and or
-    if ($_POST['provider_name'] == "") {
+    if (empty($_POST['provider_name'])) {
         $form_err .= xl('Empty value in "Name of Provider"') . '<br />';
     }
 
-    if ($_POST['message'] == "") {
+    if (empty($_POST['message'])) {
         $form_err .= xl('Empty value in "Email Text"') . '<br />';
     }
 
-    //process sql
+    // Store the new settings.  sms_gateway_type is not used for email.
+    // next_app_time, next_app_date, and notification_sent_date don't appear
+    // to be used by anyone.  notification_id is the pk, and should always
+    // be 2 for email settings.
+
     if (!$form_err) {
-        $next_app_time = $_POST['hour'] . ":" . $_POST['min'];
-        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type` ) ";
-        $sql_value = " (?, ?, ?, ?, ?, ?, ?, ?, ?) ";
-        $values = array($_POST['notification_id'], $_POST['sms_gateway_type'], $_POST['next_app_date'], $next_app_time,
-                        $_POST['provider_name'], $_POST['message'], $_POST['email_sender'], $_POST['email_subject'],
-                        $type);
+        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type`, `notification_sent_date` ) ";
+        $sql_value = " (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ";
+        $values = array($_POST['notification_id'], '', '0000-00-00', ':',
```

Comment:
we are trying to stay away from setting 0000-00-00 dates since more modern database settings won't allow this (needs to be NULL or a actual date). so, would be better to get it working with NULL:
```
$values = array($_POST['notification_id'], '', NULL, ':',
```

Reply:
Yeah, NULL would be better for sure.  I was just going with what gets seeded into the database.  We could change those fields from 'date NOT NULL' to 'date', but I think we'd also need a migration for existing databases.  Would that go into 5_0_2-to-6_0_0_upgrade.sql, or will there be another version 5 release?

As long as we're doing that, I'd also like to get rid of the columns that nobody uses (next_app_time, next_app_date, and notification_sent_date), unless you think I should leave them alone.